blaze-core/src/main/scala/org/http4s/blazecore/websocket/Http4sWSStage.scala

```diff
@@ -99,19 +100,24 @@ private[http4s] class Http4sWSStage[F[_]](
         else F.unit
       } >> deadSignal.set(true)
 
-    readFrameTrampoline.flatMap {
-      case c: Close =>
-        for {
-          s <- F.delay(sentClose.get())
-          //If we sent a close signal, we don't need to reply with one
-          _ <- if (s) deadSignal.set(true) else maybeSendClose(c)
-        } yield c
-      case p @ Ping(d) =>
-        //Reply to ping frame immediately
-        writeFrame(Pong(d), trampoline) >> F.pure(p)
-      case rest =>
-        F.pure(rest)
-    }
+    readFrameTrampoline
+      .recoverWith {
+        case _: ReservedOpcodeException => F.delay(Close(1003)).flatMap(F.fromEither)
```

Comment:
Not sure if this can be `F.fromEither(Close(1003))` directly?

Reply:
Yeah, there's no side effect here, right?

Reply:
Judging by the `apply` method, I think so.

Reply:
Done.